modules/codegen/test/src/smithy4s/codegen/internals/ScaladocSpec.scala

```diff
@@ -0,0 +1,109 @@
+package smithy4s.codegen.internals
+
+final class ScaladocSpec extends munit.FunSuite {
+  import TestUtils._
+
+  test("Generate Scaladoc - include struct field docs") {
+    val smithy =
+      """
+        |$version: "2"
+        |
+        |namespace smithy4s
+        |
+        |/// Struct docs
+        |structure DocumentedStruct {
+        |  /// int docs
+        |  @required
+        |  int: Integer
+        |  /// struct docs
+        |  struct: UndocumentedStruct
+        |  
+        |  @required
+        |  /// required before comment  
+        |  string: String
```

Comment:
Added this to highlight this problem (maybe it's just smithy synax rules in play?).
If the ordering of annotation and comment is like here (first annotation and then docs), then in AWS model, the documentation trait is missing, only `@required` trait is there. I think the same applies for other annotations, like `@packedInputs` before the comment on an operation/service etc.

Reply:
For example, in the `@packedInput` test, changing spec to 
```
@packedInputs
/// packed operation docs
operation PackedHelloOp {
  input: Hello
}
```
will make test fail

Reply:
Cool, I will replicate with minimal example, and report there. To me too it seemed like a bug, cause I couldn't find the documentation  trait anywhere else, it seemed to be just lost.

Reply:
Awesome, thanks for the heads-up, that's good to know 👍